### PR TITLE
Adding `start-docs` command and readme improvements

### PR DESCRIPTION
**Pull request by SnailBones**
## Launch Checklist

 - [ ] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Review by HarelM — APPROVED** [review]

**Comment by SnailBones** [comment]
Thanks @HarelM!

File: package.json [hunk 1]
```diff
@@ -146,6 +146,7 @@
     "generate-style-code": "npm run tsnode build/generate-style-code.ts",
     "generate-typings": "npm run tsnode build/generate-typings.ts",
     "generate-docs": "typedoc && npm run tsnode build/generate-docs.ts",
+    "start-docs": "docker run --rm -it -p 8000:8000 -v ${PWD}:/docs squidfunk/mkdocs-material",
```
Comment (HarelM) [inline package.json:149]: Can you move this command down to the start commands and keep this section for the generate commands?

File: documents_and_diagrams/README.md [hunk 2]
```diff
@@ -25,23 +25,50 @@ Examples are written as regular html files in `test/examples`. Each example shou
...
+
+```bash
+npm run generate-docs
+```
+
+Finally, run:
+
+```bash
+npm run start-docs
+```
```
Comment (HarelM) [inline documents_and_diagrams/README.md:71]: This is great, I would consider adding a note on how to update the docs while editing, i.e. keep the start-docs running and re run generate-docs to see the changes.

**Comment by SnailBones** [comment]
Thanks for the review @HarelM! Just addressed your three suggestions.

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
Thanks!!

**Pull request merged**